src/client/pages/matchPage/MatchPage.jsx

```diff
@@ -64,6 +67,28 @@ const MatchPage = () => {
             setBook(bookToMatch);
         }
     }  
+
+    const addSavedBookToUser = async (recommendedBook) => {
+      const res = await fetch(POST_RECOMMENDATION + userId, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          bookId: recommendedBook.id,
+          userId: userId,
+          isStored: true
+        })
+      });
+      const storedRecommendation = await res.json();
+      console.log('Stored book:', storedRecommendation.data);
+    }
+
+    const handleSaveClick = async (event) => {
+      const recommendedBook = library.filter(libraryBook => libraryBook.id === Number(event.target.id))[0];
```

Comment:
A nice optimisation you could make here would be to move `Number(event.target.id)` to it's own variable so this filter doesn't have to convert the ID to an int on every iteration